Generator/Sources/Generator.swift

```diff
@@ -0,0 +1,231 @@
+        var parsedAttributes = [Attribute]()
+        for element in try fileManager.subpathsOfDirectory(atPath: semConvModelsDirectory.path()) {
+            // Currently we only support parsing the `registry` files.
+            // TODO: Expand support to metric/span files, which involves handling `ref` type attributes.
+            guard element.hasSuffix("registry.yaml") || element.hasSuffix("registry-deprecated.yaml") else {
+                continue
+            }
+            let filePath = semConvModelsDirectory.appending(path: element)
+            guard
+                let fileData = fileManager.contents(atPath: filePath.path()),
+                let fileContents = String(data: fileData, encoding: .utf8)
+            else {
+                print("warning: No data found at \(filePath.path())")
+                continue
+            }
+            let file: RegistryFile
+            do {
+                file = try YAMLDecoder().decode(RegistryFile.self, from: fileContents)
+            } catch {
+                print("Error decoding \(filePath.path())")
+                throw error
+            }
+
+            for group in file.groups {
+                for groupAttribute in group.attributes {
+                    parsedAttributes.append(groupAttribute)
+                }
+            }
+        }
+        return parsedAttributes
+    }
+
+    /// Constructs a tree of namespaces from a list of attributes.
+    private func constructNamespaceTree(
+        attributes: [Attribute]
+    ) throws -> Namespace{
+        let root = Namespace(id: "")
+        for attribute in attributes {
+            let path = attribute.id.split(separator: ".")
+            var namespace = root
+            var walkedPath = [String]()
+            for pathElement in path[0 ..< (path.count - 1)] {
+                let pathElement = String(pathElement)
+                walkedPath.append(pathElement)
+
+                let nextNamespace = namespace.subNamespaces[pathElement] ?? {
+                    let newNamespace = Namespace(id: walkedPath.joined(separator: "."))
+                    namespace.subNamespaces[pathElement] = newNamespace
+                    return newNamespace
+                }()
+                namespace = nextNamespace
+            }
+            guard let attributeName = path.last.map({ String($0) }) else {
+                throw GeneratorError.attributeNameNotFound(attribute.id)
+            }
+            namespace.attributes[attributeName] = attribute
+        }
+        return root
+    }
+
+    /// Generates files based on the provided namespaces and renderers.
+    private func render(
+        fileManager: FileManager,
+        repoDirectory: URL,
+        topLevelNamespaces: [Namespace],
+        renderers: [FileRenderer]
+    ) throws {
+        for renderer in renderers {
+            let directory = repoDirectory.appending(path: "Sources/\(renderer.targetDirectory)Generated/")
```

Comment:
I think this would be easier to read if we used `appending(components: "Sources", renderer.targetDirectory, "Generated")`. Therefore, I'd also remove the included trailing slash from `renderer.targetDirectory`.

Reply:
Ah great suggestion. I implemented this change in https://github.com/swift-otel/swift-otel-semantic-conventions/pull/2/commits/b3b2c67da78f1fbe87c8646c343451a697a16bd8